### PR TITLE
Fix building libical-glib when STATIC_ONLY is enabled.

### LINKED ISSUE #527: Ninja dupbuild=err cmake -G Ninja -DICAL_GLIB=ON -DSTATIC_ONLY=ON ..
happening in the 3.0 branch

mkdir build; cd build; cmake -G Ninja -DICAL_GLIB=ON -DSTATIC_ONLY=ON ..

fails with
```
CMake Error:
  Running

   '/usr/bin/ninja-build' '-C' '/home/allen/projects/libical/libical/build-foo' '-t' 'recompact'

  failed with:

   ninja: error: build.ninja:2134: multiple rules generate lib/libical-glib.a [-w dupbuild=err]
```

### PR DESCRIPTION
**Pull request by wrobelda**
Fixes #527

### CODE REVIEW

**Comment by wrobelda** [comment]
FYI, https://github.com/microsoft/vcpkg/pull/21667

File: src/libical-glib/CMakeLists.txt [hunk 1]
```diff
@@ -200,7 +200,7 @@ if(ICAL_GLIB_VAPI)
       --girdir=${CMAKE_BINARY_DIR}/src/libical-glib
       --pkg gio-2.0
       --library libical-glib
-      --metadatadir=${CMAKE_CURRENT_SOURCE_DIR}
+      --metadatadir=${CMAKE_CURRENT_SOURCE_DIR
```
Comment (winterz) [inline src/libical-glib/CMakeLists.txt:203]: missing closing brace

Comment (wrobelda) [reply src/libical-glib/CMakeLists.txt]: Oh wow, nasty typo. Fixed, thanks!

**Comment by winterz** [comment]
with this patch (and fixing the missing closing brace by-hand) I'm seeing lots of unresolved symbols

[```
135/135] Generating libical-glib documentation
FAILED: doc/reference/libical-glib/html/index.html 
...
/home/allen/projects/libical/libical/build-testninjagcc7-ninjagcc/lib/libical-glib.a(i-cal-array.c.o):i-cal-array.c:function i_cal_array_new_full: error: undefined reference to 'icalarray_free'
/home/allen/projects/libical/libical/build-testninjagcc7-ninjagcc/lib/libical-glib.a(i-cal-array.c.o):i-cal-array.c:function i_cal_array_copy: error: undefined reference to 'icalarray_copy'
/home/allen/projects/libical/libical/build-testninjagcc7-ninjagcc/lib/libical-glib.a(i-cal-array.c.o):i-cal-array.c:function i_cal_array_free: error: undefined reference to 'icalarray_free'
... and lots more
```

**Comment by wrobelda** [comment]
For what it's worth, I apply it against 3.0.11 source. Is that what you're testing with?

**Comment by winterz** [comment]
the linking error is unrelated.  thanks for this patch @wrobelda

**Pull request merged**